File: frontend/src/api/ai-mcp.ts
/**
 * MCP 服务管理 API
 *
 * 命名约定：本文件所有接口字段与服务端保持一致（snake_case），
 * 后端 Pydantic schema 不做驼峰转换，axios 拦截器也不做字段重命名。
 * 如需驼峰字段，请在调用侧手动映射。
 */
import request from '@/utils/request'

// ========== Types ==========

export interface McpApiKey {
  id: number
  name: string
  service_type: string
  platform?: string | null
  protocol_type?: string | null
  version?: string | null
  description?: string | null
  service_url?: string | null
  service_name?: string | null
  api_key?: string | null
  namespace?: string | null
  group_key?: string | null
  access_path?: string | null
  properties?: Record<string, any> | null
  capabilities?: string[] | null
  remark?: string | null
  status: number
  sort: number
  template_id: number
  health_status: string
  last_check_at?: string | null
  creator?: string | null
  updater?: string | null
  created_at?: string
  updated_at?: string
}

export interface McpClient {
  id: number
  name: string
  api_key_id: number
  client_type: string
  mcp_type: string
  version?: string | null
  description?: string | null
  tools_config?: Record<string, any> | null
  remark?: string | null
  status: number
  creator?: string | null
  created_at?: string
}

export interface McpSquareTemplate {
  id: number
  name: string
  icon?: string | null
  category?: string | null
  platform?: string | null
  description?: string | null
  service_type: string
  service_url?: string | null
  access_path?: string | null
  version?: string | null
  capabilities?: string[] | null
  default_client_config?: Record<string, any> | null
  sort: number
  status?: number
  is_installed: boolean
  created_at?: string
  updated_at?: string
}

export interface PageResp<T> {
  data: T[]
  total: number
  page: number
  pageSize: number
}

// ========== API Key ==========

export function getMcpApiKeyPage(params: {
  name?: string
  service_type?: string
  status?: number
  page?: number
  pageSize?: number
}) {
  return request.get<PageResp<McpApiKey>>('/api/v1/admin/ai/mcp-api-key/page', { params })
}

export function getMcpApiKeyDetail(id: number) {
  return request.get<McpApiKey>('/api/v1/admin/ai/mcp-api-key/get', { params: { id } })
}

export function createMcpApiKey(data: Partial<McpApiKey>) {
  return request.post<{ id: number; message: string }>('/api/v1/admin/ai/mcp-api-key/create', data)
}

export function updateMcpApiKey(data: Partial<McpApiKey> & { id: number }) {
  return request.post<{ id: number; message: string }>('/api/v1/admin/ai/mcp-api-key/update', data)
}

export function deleteMcpApiKey(id: number) {
  return request.delete<{ message: string }>('/api/v1/admin/ai/mcp-api-key/delete', { params: { id } })
}

export function getMcpApiKeySimpleList() {
  return request.get<{ id: number; name: string; service_type: string }[]>(
    '/api/v1/admin/ai/mcp-api-key/simple-list'
  )
}

export interface McpServerSelectOption {
  value: number | string
  label: string
  service_type: string
  source: 'installed' | 'builtin'
  template_id: number
  /** 兼容回填逻辑补充的字段（实际列表中会以 label 填充） */
  name?: string
  description?: string
}

export function getMcpServerSelect() {
  return request.get<McpServerSelectOption[]>(
    '/api/v1/admin/ai/mcp-api-key/simple-select'
  )
}

// ========== Client ==========

export function getMcpClientPage(params: {
  api_key_id: number
  page?: number
  pageSize?: number
}) {
  return request.get<PageResp<McpClient>>('/api/v1/admin/ai/mcp-client/page', { params })
}

export function getMcpClientDetail(id: number) {
  return request.get<McpClient>('/api/v1/admin/ai/mcp-client/get', { params: { id } })
}

export function createMcpClient(data: Partial<McpClient>) {
  return request.post<{ id: number; message: string }>('/api/v1/admin/ai/mcp-client/create', data)
}

export function updateMcpClient(data: Partial<McpClient> & { id: number }) {
  return request.post<{ id: number; message: string }>('/api/v1/admin/ai/mcp-client/update', data)
}

export function deleteMcpClient(id: number) {
  return request.delete<{ message: string }>('/api/v1/admin/ai/mcp-client/delete', { params: { id } })
}

// ========== Square ==========

export function getMcpSquarePage(params: {
  name?: string
  category?: string
  status?: number
  page?: number
  pageSize?: number
}) {
  return request.get<PageResp<McpSquareTemplate>>('/api/v1/admin/ai/mcp-square/page', { params })
}

export function getMcpSquareDetail(id: number) {
  return request.get<McpSquareTemplate>('/api/v1/admin/ai/mcp-square/get', { params: { id } })
}

export function createMcpSquare(data: Partial<McpSquareTemplate>) {
  return request.post<{ id: number; message: string }>('/api/v1/admin/ai/mcp-square/create', data)
}

export function updateMcpSquare(data: Partial<McpSquareTemplate> & { id: number }) {
  return request.post<{ id: number; message: string }>('/api/v1/admin/ai/mcp-square/update', data)
}

export function deleteMcpSquare(id: number, force = false) {
  return request.delete<{ message: string }>('/api/v1/admin/ai/mcp-square/delete', {
    params: { id, force },
  })
}

export interface McpSquareInstallData {
  template_id: number
  name?: string
  service_url?: string
  api_key?: string
  access_path?: string
  service_name?: string
  namespace?: string
  group_key?: string
}

export function installMcpSquare(data: McpSquareInstallData) {
  return request.post<{ id: number; message: string }>('/api/v1/admin/ai/mcp-square/install', data)
}

export function testMcpConnection(id: number) {
  return request.post<{ id: number; health_status: string; detail: string; checked_at: string }>(
    `/api/v1/admin/ai/mcp-square/test/${id}`
  )
}

export function uninstallMcpSquare(apiKeyId: number) {
  return request.delete<{ message: string }>('/api/v1/admin/ai/mcp-square/uninstall', {
    params: { apiKeyId },
  })
}

// ========== MCP 已注册工具 ==========

export interface McpRegisteredTool {
  name: string
  description: string
  category: string
  input_schema: Record<string, any>
}

export function getMcpToolsList(params: {
  name?: string
  category?: string
  page?: number
  pageSize?: number
}) {
  return request.get<PageResp<McpRegisteredTool>>('/api/v1/admin/ai/mcp-tools/list', { params })
}

File: backend/app/schemas/ai/ai_mcp.py
"""MCP 服务管理 Schemas - API Key / Client / Square。"""
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, ConfigDict, Field


# ============= AiMcpApiKey Schemas =============

class AiMcpApiKeyCreate(BaseModel):
    """创建 MCP API Key"""
    name: str = Field(..., description="密钥名称")
    service_type: str = Field(..., description="服务类型")
    platform: Optional[str] = None
    version: Optional[str] = None
    description: Optional[str] = None
    service_url: Optional[str] = None
    service_name: Optional[str] = None
    api_key: Optional[str] = None
    namespace: Optional[str] = None
    group_key: Optional[str] = None
    access_path: Optional[str] = None
    properties: Optional[dict] = None
    capabilities: Optional[List[str]] = None
    remark: Optional[str] = None


class AiMcpApiKeyUpdate(BaseModel):
    """更新 MCP API Key"""
    id: int = Field(..., description="ID")
    name: Optional[str] = None
    service_type: Optional[str] = None
    platform: Optional[str] = None
    version: Optional[str] = None
    description: Optional[str] = None
    service_url: Optional[str] = None
    service_name: Optional[str] = None
    api_key: Optional[str] = None
    namespace: Optional[str] = None
    group_key: Optional[str] = None
    access_path: Optional[str] = None
    properties: Optional[dict] = None
    capabilities: Optional[List[str]] = None
    remark: Optional[str] = None
    status: Optional[int] = None
    sort: Optional[int] = None


class AiMcpApiKeyPageResp(BaseModel):
    """MCP API Key 分页响应"""
    id: int
    name: str
    service_type: str
    platform: Optional[str] = None
    protocol_type: Optional[str] = None
    version: Optional[str] = None
    description: Optional[str] = None
    status: int
    sort: int
    template_id: int = 0
    health_status: str = "unknown"
    last_check_at: Optional[datetime] = None
    creator: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class AiMcpApiKeyDetailResp(AiMcpApiKeyPageResp):
    """MCP API Key 详情响应"""
    service_url: Optional[str] = None
    service_name: Optional[str] = None
    api_key: Optional[str] = None
    namespace: Optional[str] = None
    group_key: Optional[str] = None
    access_path: Optional[str] = None
    properties: Optional[dict] = None
    capabilities: Optional[List[str]] = None
    remark: Optional[str] = None
    updater: Optional[str] = None


class AiMcpApiKeySimpleResp(BaseModel):
    """MCP API Key 简易响应（用于下拉选择）"""
    id: int
    name: str
    service_type: str

    model_config = ConfigDict(from_attributes=True)


# ============= McpClient Schemas =============

class AiMcpClientCreate(BaseModel):
    """创建 MCP Client"""
    name: str = Field(..., description="客户端名称")
    api_key_id: int = Field(..., description="关联API Key ID")
    client_type: str = Field(..., description="客户端类型")
    mcp_type: str = Field(..., description="应用类别")
    version: Optional[str] = None
    description: Optional[str] = None
    tools_config: Optional[dict] = None
    remark: Optional[str] = None


class AiMcpClientUpdate(BaseModel):
    """更新 MCP Client"""
    id: int = Field(..., description="ID")
    name: Optional[str] = None
    client_type: Optional[str] = None
    mcp_type: Optional[str] = None
    version: Optional[str] = None
    description: Optional[str] = None
    tools_config: Optional[dict] = None
    remark: Optional[str] = None
    status: Optional[int] = None


class AiMcpClientPageResp(BaseModel):
    """MCP Client 分页响应"""
    id: int
    name: str
    api_key_id: int
    client_type: str
    mcp_type: str
    version: Optional[str] = None
    description: Optional[str] = None
    status: int
    creator: Optional[str] = None
    created_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


# ============= McpSquare Schemas =============

class McpSquareInstallReq(BaseModel):
    """MCP 广场安装请求 - 模板ID + 用户自定义参数"""
    template_id: int = Field(..., description="模板ID")
    name: Optional[str] = Field(None, description="自定义名称（默认用模板名）")
    service_url: Optional[str] = Field(None, description="用户自定义服务地址")
    api_key: Optional[str] = Field(None, description="用户鉴权密钥")
    access_path: Optional[str] = Field(None, description="自定义访问路径")
    service_name: Optional[str] = Field(None, description="Nacos服务名")
    namespace: Optional[str] = Field(None, description="Nacos命名空间")
    group_key: Optional[str] = Field(None, description="Nacos分组")


class McpSquareCreate(BaseModel):
    """创建 MCP 广场模板"""
    name: str = Field(..., max_length=100, description="模板名称")
    icon: Optional[str] = Field(None, max_length=200, description="图标标识")
    category: Optional[str] = Field(None, max_length=50, description="分类")
    platform: Optional[str] = Field(None, max_length=50, description="平台类型")
    description: Optional[str] = Field(None, description="描述")
    service_type: str = Field(..., max_length=20, description="服务类型")
    service_url: Optional[str] = Field(None, max_length=500, description="服务地址模板")
    access_path: Optional[str] = Field(None, max_length=500, description="访问路径")
    version: Optional[str] = Field(None, max_length=20, description="协议版本")
    capabilities: Optional[List[str]] = None
    default_client_config: Optional[dict] = None
    sort: int = Field(0, description="排序")
    status: int = Field(1, description="1=启用,0=禁用")


class McpSquareUpdate(BaseModel):
    """更新 MCP 广场模板"""
    id: int = Field(..., description="ID")
    name: Optional[str] = Field(None, max_length=100)
    icon: Optional[str] = Field(None, max_length=200)
    category: Optional[str] = Field(None, max_length=50)
    platform: Optional[str] = Field(None, max_length=50)
    description: Optional[str] = None
    service_type: Optional[str] = Field(None, max_length=20)
    service_url: Optional[str] = Field(None, max_length=500)
    access_path: Optional[str] = Field(None, max_length=500)
    version: Optional[str] = Field(None, max_length=20)
    capabilities: Optional[List[str]] = None
    default_client_config: Optional[dict] = None
    sort: Optional[int] = None
    status: Optional[int] = None


class McpSquarePageResp(BaseModel):
    """MCP 广场模板分页响应"""
    id: int
    name: str
    icon: Optional[str] = None
    category: Optional[str] = None
    platform: Optional[str] = None
    description: Optional[str] = None
    service_type: str
    service_url: Optional[str] = None
    access_path: Optional[str] = None
    version: Optional[str] = None
    capabilities: Optional[List[str]] = None
    default_client_config: Optional[dict] = None
    sort: int
    status: int = 1
    is_installed: bool = False
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)
